File: StaticSharp/Modifiers/UserSelect/UserSelect.cs
using StaticSharp.Gears;
using System.Runtime.CompilerServices;

namespace StaticSharp {


    public enum UserSelectOption {
        None,
        Text,
        All,
        Auto
    }

    public interface JUserSelect : JModifier {
        public UserSelectOption Option { get; set; }
    }


    [ConstructorJs]
    public partial class UserSelect : Modifier {
        public UserSelect(UserSelectOption option, [CallerLineNumber] int callerLineNumber = 0, [CallerFilePath] string callerFilePath = "")
            : this(callerLineNumber, callerFilePath) {
            Option = option;
        }
    }


}

File: StaticSharp/Modifiers/UserSelect/UserSelect.js
StaticSharpClass("StaticSharp.UserSelect", (modifier, element) => {
    StaticSharp.Modifier(modifier, element)

    modifier.Reactive = {
        Option: "None"
    }

    new Reaction(() => {
        var option = modifier.Option
        
        if ((option === undefined) || (option === ""))
            element.style.userSelect = ""
        else
            element.style.userSelect = CamelToKebab(option);
    })
})